src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const { id, title, completed, userId } = currentTodo ?? {};
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (userId) {
+      getUser(userId)
+        .then(setUser)
+        .finally(() => setLoading(false));
+    }
+  }, [userId]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loading && <Loader />}
```

Comment:
The `Loader` component is conditionally rendered based on the `loading` state. Ensure that the `Loader` component is correctly imported and implemented to avoid runtime errors.